src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeQuality;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeStructFieldsWritable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : AbstractCodeQualityDiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
+            IDEDiagnosticIds.MakeStructFieldsWritable,
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            isUnneccessary: false);
+
+        public CSharpMakeStructFieldsWritableDiagnosticAnalyzer()
+            : base(ImmutableArray.Create(s_diagnosticDescriptor), GeneratedCodeAnalysisFlags.ReportDiagnostics)
```

Comment:
this constructor seems more convoluted than most of our DiagnosticAnalyzer constructors in teh IDE.  I would expect this to just call into this base constructor:

```c#
protected AbstractCodeStyleDiagnosticAnalyzer(
            string descriptorId, LocalizableString title,
            LocalizableString messageFormat = null,
            bool configurable = true)
```

Reply:
There is no such constructor in `AbstractCodeQualityDiagnosticAnalyzer`, or at least I'm not able to find it.